packages/twenty-server/src/engine/metadata-modules/field-metadata/interfaces/field-metadata-settings.interface.ts

```diff
@@ -12,6 +12,7 @@ type FieldMetadataDefaultSettings = {
 
 type FieldMetadataNumberSettings = {
   dataType: NumberDataType;
+  decimals?: number;
```

Comment:
@gitstart-twenty for the record you can tests api requests visiting
http://localhost:3000/metadata
Then just get the Authorization header from any request in the product to load the workspace schema. 
You can build and run your own queries and mutations. You can also c/c them from the requests in the product to reproduce them:

https://github.com/user-attachments/assets/73054fd1-1a99-49ab-8d9a-7b2c7af8c14e

Reply:
@gitstart-twenty I think it would be a good idea to add a settings validation service that we could call in fieldMetadataService in the existing validateFieldMetadataInput: 

```
private validateFieldMetadataInput<
    T extends UpdateFieldInput | CreateFieldInput,
  >(fieldMetadataInput: T, objectMetadata: ObjectMetadataEntity): T {
    if (fieldMetadataInput.name) {
       [...]
    }

    if (fieldMetadataInput.options) {
      [...]
    }

    **if (fieldMetadataInput.settings) {
      this.fieldMetadataValidationService.validateSettingsOrThrow(
        fieldMetadataInput.type,
        fieldMetadataInput.settings,
      );
    }**

    return fieldMetadataInput;
  } 
```

Reply:
Sure, thanks for the suggestion